tests/algorithms/component_connection_tests_cuda.cpp

```diff
@@ -0,0 +1,91 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ * 
+ * (c) 2021 CERN for the benefit of the ACTS project
+ * 
+ * Mozilla Public License Version 2.0
+ */
+
+#include "vecmem/containers/array.hpp"
+#include "vecmem/containers/vector.hpp"
+#include "vecmem/memory/cuda/device_memory_resource.hpp"
+#include "vecmem/memory/cuda/host_memory_resource.hpp"
+#include "vecmem/memory/cuda/managed_memory_resource.hpp"
+#include "vecmem/utils/cuda/copy.hpp"
+
+#include "clusterization/component_connection_kernels.cuh"
+#include <gtest/gtest.h>
+
+// This defines the local frame test suite
+TEST(algorithms, component_connection_cuda){
+
+  std::vector<traccc::cell> cell_items = 
+    { {1, 0, 1., 0. }, 
+      {8, 4, 2., 0.}, 
+      {10, 4, 3., 0.}, 
+      {9, 5, 4., 0.}, 
+      {10, 5, 5., 0}, 
+      {12, 12, 6, 0}, 
+      {3, 13, 7, 0}, 
+      {11, 13, 8, 0}, 
+      {4, 14, 9, 0 } };
+
+  vecmem::cuda::managed_memory_resource managed_resource;
+  
+  vecmem::vector< traccc::cell > cell_per_module(&managed_resource);
+  vecmem::jagged_vector< traccc::cell > cell_per_event(&managed_resource);
+
+  // move cell_items to cell_per_module
+  for (auto i=0; i<cell_items.size(); i++){
+    traccc::cell aCell;
+    aCell.channel0 = std::move(cell_items[i].channel0);
+    aCell.channel1 = std::move(cell_items[i].channel1);
+    aCell.activation = std::move(cell_items[i].activation);
+    aCell.time = std::move(cell_items[i].time);
+    cell_per_module.push_back(aCell);
+  }
+
+  vecmem::vector< unsigned int > label_per_module(cell_per_module.size(),0,&managed_resource);
+  vecmem::jagged_vector< unsigned int > label_per_event(&managed_resource);
+  
+  // fill jagged_vector
+  cell_per_event.push_back(cell_per_module);
+  label_per_event.push_back(label_per_module);
+
+  vecmem::vector< unsigned int> num_labels(cell_per_event.size(), 0 , &managed_resource);
+
+  // run sparse_ccl
+  vecmem::data::jagged_vector_data< traccc::cell> cell_data(cell_per_event,&managed_resource);
+  vecmem::data::jagged_vector_data< unsigned int > label_data(label_per_event, &managed_resource);
```

Comment:
You could simplify these calls to:

```c++
auto cell_data = vecmem::get_data( cell_per_event, &managed_resource );
auto label_data = vecmem::get_data( label_per_event, &managed_resource );
```

At least as long as you use the "managed resource". For explicit memory movements the code will unfortunately need to become a bit more elaborate. But that's something that we're still in the process of figuring out ourselves... (https://github.com/acts-project/vecmem/pull/60)

Reply:
Is that right? I got the following error: 
```
error: no matching function for call to ‘get_data(vecmem::data::jagged_vector_data<unsigned int>&, vecmem::cuda::managed_memory_resource*)’
```

Reply:
The `get_data` method is designed to work with jagged standard vectors, so either `std::vector<std::vector<T>>` or `vecmem::vector<vecmem::vector<T>>`. If you already have a `jagged_vector_data` there should be no reason to use them. :slightly_smiling_face:

Reply:
I'm confused though. 😦  The code that we see here has `cell_per_event` and `label_per_event` both declared as `vecmem::jagged_vector` types. So I'm not sure how you ended up calling `vecmem::get_data(...)` on a `vecmem::data::jagged_vector_data` type here... 😕